web/src/drafts.js

```diff
@@ -274,6 +275,62 @@ export function update_draft(opts = {}) {
 
 export const DRAFT_LIFETIME = 30;
 
+export function get_draft_ids_for_topic(stream_id, topic) {
+    const drafts = draft_model.get();
+    return Object.keys(drafts)
+        .filter(
+            (draft_id) =>
+                drafts[draft_id].type === "stream" &&
+                drafts[draft_id].stream_id === stream_id &&
+                drafts[draft_id].topic === topic,
+        )
+        .sort((draft_a, draft_b) => drafts[draft_a].updatedAt - drafts[draft_b].updatedAt);
+}
+
+export function get_draft_ids_for_private_message_recipient(private_message_recipient) {
+    const drafts = draft_model.get();
+    return Object.keys(drafts)
+        .filter(
+            (draft_id) =>
+                drafts[draft_id].type === "private" &&
+                _.isEqual(
+                    drafts[draft_id].private_message_recipient.split(",").sort(),
+                    private_message_recipient.split(",").sort(),
+                ),
```

Comment:
Is the splitting/sorting the thing to do here? I would have thought that the `private_message_recipient` fields were already a sorted, comma-separated list of `user_id` values, and so comparing as strings should have the exact same result.